components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java

```diff
@@ -3431,7 +3431,7 @@
                 " INSERT INTO AM_DEPLOYMENT_REVISION_MAPPING (NAME, VHOST, REVISION_UUID, DISPLAY_ON_DEVPORTAL)" +
                         " VALUES (?,?,?,?)";
         public static final String ADD_DEPLOYED_API_REVISION =
-                " INSERT INTO AM_DEPLOYED_REVISION (NAME, VHOST, REVISION_UUID)" +
+                " REPLACE INTO AM_DEPLOYED_REVISION (NAME, VHOST, REVISION_UUID)" +
```

Comment:
If we are using `REPLACE` do we need line https://github.com/wso2/carbon-apimgt/pull/10698/files#diff-19a032cb823fd4f31e79376db6427e305b79949a26979f42d5e7ea0c607cdff2R8802?  i.e ` apiMgtDAO.removeDeployedAPIRevision(apiId, environmentsToRemove);`

Reply:
it's required. because there might be old records with different revisionuuids for the same API in same env. replace won't catch it because of the different revisionUUID

Reply:
lets change this to remove and insert model

Reply:
We did the removing before adding from here even earlier. https://github.com/wso2/carbon-apimgt/pull/10698/files#diff-19a032cb823fd4f31e79376db6427e305b79949a26979f42d5e7ea0c607cdff2R8802. However I noticed there were intermittent error logs saying "duplicate entries violating primary key constraint." I suspected that it can be due to duplicate entries in the same batch. Hence added duplicates removing logic. To be on the safe side added "replace" here as well.  I guess we can keep it as insert since we are removing duplicates from the message.

Reply:
Lets not use "REPLACE" as it can swallow and difficult to find inconsistencies that can happen.

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/10698/commits/661bcb9e2f6b0a28662af8e1acc8d42bab5af7e0